src/main/java/org/opentripplanner/graph_builder/linking/SampleStopLinker.java

```diff
@@ -0,0 +1,131 @@
+package org.opentripplanner.graph_builder.linking;
+
+import java.util.Collection;
+
+import gnu.trove.map.TIntObjectMap;
+import gnu.trove.map.hash.TIntIntHashMap;
+import gnu.trove.map.hash.TIntObjectHashMap;
+
+import org.mapdb.Fun.Tuple2;
+import org.opentripplanner.analyst.core.Sample;
+import org.opentripplanner.analyst.request.SampleFactory;
+import org.opentripplanner.common.geometry.GeometryUtils;
+import org.opentripplanner.common.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.routing.edgetype.IntersectionTransitLink;
+import org.opentripplanner.routing.edgetype.SimpleTransfer;
+import org.opentripplanner.routing.edgetype.StreetTransitLink;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.vertextype.OsmVertex;
+import org.opentripplanner.routing.vertextype.TransitStop;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Multimap;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+
+/**
+ * Link transit stops to the street network in a non-destructive manner; i.e. don't
+ * modify the street network. Being non-destructive is important in Analyst; new transit
+ * models should not affect the length of existing streets, even slightly (due to float
+ * roundoff from splits, etc.) An error of even a few seconds can be significant if it causes
+ * the router to miss a transit vehicle or transfer, or causes the stop to move just beyond a
+ * hard boundary. 
+ * 
+ * We do this by using the same linking code that is used
+ * for Analyst (the SampleFactory code) which additionally means that this code
+ * path is used and tested in multiple applications.
+ */
+public class SampleStopLinker {
```

Comment:
If I understand right this is not currently used. But it could be very useful for patching in temporary lines in interactively generated scenarios.